### PR TITLE
feat(integrations): Add support for text input with choices

### PR DESCRIPTION
**Pull request by dcramer**

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/static/sentry/app/components/forms/select2Field.jsx [hunk 1]
```diff
@@ -48,6 +66,12 @@ class Select2Field extends InputField {
       allowClear: this.props.allowClear,
       allowEmpty: this.props.allowEmpty,
       width: 'element',
+      data: (this.props.choices || []).map(choice => {
+        if (choice.constructor === Array) {
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/forms/select2Field.jsx]: `Array.isArray`

**Review by billyvg — COMMENTED** [review]
Looks good but can we add some unit tests?

File: src/sentry/static/sentry/app/components/forms/select2Field.jsx (hunk 1 above)
Comment (dcramer) [reply src/sentry/static/sentry/app/components/forms/select2Field.jsx]: SO suggested this was a faster method, and I was concerned about 'instanceof' for compatibility reasons

Comment (billyvg) [reply src/sentry/static/sentry/app/components/forms/select2Field.jsx]: `Array.isArray` is faster on Chrome 61 at least:
https://jsperf.com/array-isarray-vs-instanceof-vs-constructor-check

**Comment by dcramer** [comment]
Updated with an array of tests

**Review by billyvg — APPROVED** [review]

**Pull request merged**